draft-ietf-quic-transport.md

```diff
@@ -1069,6 +1069,15 @@ to cease using the connection IDs when requested can result in connection
 failures, as the issuing endpoint might be unable to continue using the
 connection IDs with the active connection.
 
+An endpoint SHOULD limit the number of in flight RETIRE_CONNECTION_ID frames
+to bound the necessary state. In order to minimize delay in common situations,
+the limit on the number of in flight RETIRE_CONNECTION_IDs SHOULD be at least
+the active_connection_id_limit. An endpoint MAY choose to treat having too many
```

Comment:
For the first two sentences, I think we should better use phrases like "state required to track the CIDs being retired." I also think that we should provide guidance on choosing the limit, which should be greater than active_connection_id_limit (see #3509).

"in flight" is confusing, as people might read it as if it excludes sequence number of retired CIDs that are to be sent but cannot be sent immediately. There is no guarantee that RCID frames for all the sequence numbers to be retired can be sent immediately as the frame is under the control of CC. Note that there would be an incentive for people to consider that "in flight" here excludes those yet to be sent, because is the next setence we talk about "having too many in need of retirement." That gives the impression that "in flight" means something narrower.

Reply:
it does exclude ones that can't be sent; the point is, even if there is space in the cwnd I can choose to send fewer to simplify the implementation.

Reply:
If it excludes those that can't be sent, then such an endpoint is vulnerable to the attack. Unless the amount of state that is used to track those that can't be sent is bounded. We need to recommend endpoints to limit _this_ state (see https://github.com/quicwg/base-drafts/pull/3547#discussion_r401984660), not the number of RCID frames sent at once.